Sources/SKCore/CompilationDatabase.swift

```diff
@@ -185,12 +197,18 @@ extension JSONCompilationDatabase: Codable {
 }
 
 extension JSONCompilationDatabase {
-  public init(directory: AbsolutePath, _ fileSystem: FileSystem = localFileSystem) throws {
+  public init?(directory: AbsolutePath, _ fileSystem: FileSystem = localFileSystem) throws {
```

Comment:
Previously, this would throw an error both if the file was missing or if parsing failed. I updated the behavior to simply return `nil` if the file does not exist, allowing me to more correctly implement the search behavior in `tryLoadCompilationDatabase` above. This makes things slightly more strict than they were, in the unlikely case that both types of database are present _and_ the JSON fails to parse this will result in an error rather than falling back to compile_flags.txt

Reply:
That makes sense to me. Could you add a doc-comment to the initializer explaining when this returns `nil` and when it throws?